### DISCUSSION TITLE
Question about UNet1DConditionModel #6522

### DISCUSSION THREAD

**Discussion by boqian-li** [huggingface/diffusers#6538] [Ideas]
I am working on processing 1-D data with conditions using diffusion, and I am using Hugging Face models to accelerate my work. I am using UNet2DConditionModel, but the kernels are all in the shape of (3, 3), and I need to pad my 1-D data with padding=1, which means the upper (1, 3) and the lower (1, 3) in the kernels are useless and wasting lots of computing resources. I want to know if there is any good solution for my work (UNet model for conditional 1-D). I tried to modify the code in diffusers, but there are too many Conv.

**Comment by patrickvonplaten** [comment]
What exactly do you mean by "there are too many Conv"?

**Reply by boqian-li** [reply]
I mean it's very hard for users to modify the whole model to 1dConditionModel theirselves. It's a huge work

**Comment by maximilienleclei** [comment]
hi any update on a potential `UNet1DConditionModel` implementation?
